lib/register.php

```diff
@@ -407,10 +407,18 @@ function gutenberg_register_post_types() {
 function gutenberg_register_rest_routes() {
 	$controller = new WP_REST_Reusable_Blocks_Controller();
 	$controller->register_routes();
+
+	foreach ( get_post_types( array(), 'objects' ) as $post_type_object ) {
+		if( ! empty( $post_type_object->rest_base ) ) {
+			if ( post_type_supports( $post_type_object->name, 'revisions' ) ) {
+				$autosaves = new WP_REST_Autosaves_Controller( $post_type_object->name );
```

Comment:
Can I ask why do we need A REST controller dependent on the post type? Can't we use a generic controller with the post type as part of the URLs?

Reply:
To keep the code in my controller minimal, I extended the revisions controller which works off object type routes, eg. posts/[ID]/revisions/[ID] - autosaves are a special type of revision and always have a 'post parent' just like revisions. not all post types support revisions or autosaves.